qsimcirq/qsim_circuit.py

```diff
@@ -187,11 +187,21 @@ def translate_cirq_to_qsim(
     # qsim numbers qubits in reverse order from cirq
     ordered_qubits = list(reversed(ordered_qubits))
 
+    def has_qsim_kind(op: cirq.ops.GateOperation):
+      return _cirq_gate_kind(op.gate) != None
```

Comment:
[No action required]  `_cirq_gate_kind` is called here and below (line 215). Would it be possible to call it only once?

Reply:
This is just explicitly defining the `lambda` that was previously in the `cirq.decompose` call. Due to the required return types for `cirq.decompose`, it can't really be re-written to fit both use cases.

Reply:
I think that, for each gate, `_cirq_gate_kind` is called in line 204 (implicitly via the `lambda` function) and one more time in line 215. I was wondering if it is possible to modify the code in such a way that `_cirq_gate_kind` is called only once for each gate.

Reply:
Ah, sorry for the confusion. `cirq.decompose` mutates the circuit and may call `_cirq_gate_kind` multiple times for each gate. There may be a solution involving memoization, but it would require changes to `cirq.decompose` which aren't really feasible here. Thanks for checking, though!